File: PythonAlgorithmInterView/선형자료구조/배열파티션1.py
def arrayPairSum(self, nums: list[int]) -> int:

    min_arr = []
    sum = 0
    nums.sort()


    for i in range(len(nums)):

        min_arr.append(nums[i])

        if len(min_arr) == 2 :
            sum += min(min_arr)
            min_arr = []

    print(sum)




if __name__ == '__main__':
    nums = [6,2,6,5,1,2]

    print(arrayPairSum(None, nums))

File: PythonAlgorithmInterView/선형자료구조/배열파티션1.js
/**
 * @param {number[]} nums
 * @return {number}
 * 561번 문제.
 */
 var arrayPairSum = function(nums) {

    let minArr = []
    let sum = 0;

    nums.sort((a, b)=> a-b)

    nums.forEach((num)=> {
        minArr.push(num)

        if(minArr.length == 2){
            sum += Math.min(minArr[0], minArr[1]);
            minArr = []
        }
    })
    return sum
};

let nums = [1,4,3,2];
console.log(arrayPairSum(nums))